src/serena/util/file_system.py

```diff
@@ -1,20 +1,18 @@
 import os
-from collections.abc import Sequence
+from collections.abc import Callable
 
 
 def scan_directory(
     path: str,
     recursive: bool = False,
     relative_to: str | None = None,
-    ignored_dirs: Sequence[str] = (),
-    ignored_files: Sequence[str] = (),

```

Comment:
should we not also have is_ingored_file here? If feel like it would be more convenient than having to filter afterwards

Reply:
Depends on what we want the tool to do.
The original idea was: List everything except sub-directories that are explicitly ignored.
I restored this behaviour.

The previous behaviour was: Ignore everything the language server ignores (in particular, all non-source files).

If we want to ignore files, we could think about applying the new option I added and define
```
is_ignored_file = lambda p: language_server.is_ignored_path(p, ignore_unsupported_files=False)
```
which would ignore only files that are defined in `.gitignore` but keep non-source files otherwise.

Reply:
The option I added basically makes `get_ignore_spec` obsolete, and I think we should get rid of it, as it is easy to use it incorrectly.

Reply:
Yes, I think that's a good default for is_ignored_file

Reply:
Add this in the second commit.